### PR TITLE
ruby ninja

### PR DESCRIPTION
**Pull request by mandarbapaye**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/mandarbapaye/git-play/pull/6" target="_blank">ReviewNinja

File: ninja.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        @memoized_locales ||= super
+      end
+
+      def store_translations(locale, data, options = {})
+        reset_memoizations!(locale)
+        super
+      end
+
+      def reload!
+        reset_memoizations!
```
Comment (mandar-masterclass) [inline ninja.rb]: not ideal. dont reset every time.

File: ninja.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+        def lookup(locale, key, scope = nil, options = {})
+          flat_key  = I18n::Backend::Flatten.normalize_flat_keys(locale,
+            key, scope, options[:separator]).to_sym
+          flat_hash = memoized_lookup[locale.to_sym]
+          flat_hash.key?(flat_key) ? flat_hash[flat_key] : (flat_hash[flat_key] = super)
+        end
+
+        def memoized_lookup
+          @memoized_lookup ||= Hash.new { |h, k| h[k] = {} }
```
Comment (mandar-masterclass) [inline ninja.rb:29]: Good idea.

File: ninja.rb (hunk 1 above)
Comment (mandar-masterclass) [inline ninja.rb]: !fix

Comment (mandarbapaye) [inline ninja.rb]: Why not reset ? :cry:

File: ninja.rb (hunk 2 above)
Comment (mandarbapaye) [inline ninja.rb:29]: !fixed

File: ninja.rb (hunk 1 above)
Comment (mandarbapaye) [inline ninja.rb]: ok. i will do it

File: ninja.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        @memoized_locales ||= super
+      end
+
+      def store_translations(locale, data, options = {})
+        reset_memoizations!(locale)
+        super
+      end
+
+      def reload!
+        reset_memoizations!(false)
```
Comment (mandar-masterclass) [inline ninja.rb:14]: No No No !resolve

**Pull request closed without merging**